c/tskit/tables.c

```diff
@@ -627,6 +627,20 @@ write_metadata_schema_header(
     return fprintf(out, fmt, (int) metadata_schema_length, metadata_schema);
 }
 
+int
+tsk_reference_sequence_free(tsk_reference_sequence_t *self)
+{
+    if (self != NULL) {
```

Comment:
We're checking this already in table_collection_free. Best to stick with the standard idiom of assuming self is not NULL.

Reply:
Done